backend/api/src/main/kotlin/io/tolgee/api/v2/controllers/NotificationController.kt

```diff
@@ -37,8 +36,33 @@ class NotificationController(
   @AllowApiAccess
   fun getNotifications(
     @ParameterObject pageable: Pageable,
-  ): PagedModel<NotificationModel> {
+  ): NotificationPagedModel {
     val notifications = notificationService.getNotifications(authenticationFacade.authenticatedUser.id, pageable)
-    return pagedResourcesAssembler.toModel(notifications, NotificationModelAssembler(enhancers, notifications))
+    val unseenCount =
+      notificationService.getCountOfUnseenNotifications(authenticationFacade.authenticatedUser.id)
+    val pagedNotifications = pagedResourcesAssembler.toModel(notifications, NotificationModelAssembler(enhancers, notifications))
+    return NotificationPagedModel.of(pagedNotifications, unseenCount)
+  }
+
+  @PutMapping("/mark-seen")
```

Comment:
This is not correctly restful. The `/v2/notifications/{id}` has to be reserved for manipulating notification with specific id. You will have to change it so the url is `/v2/mark-notifications-seen`.